torch/_decomp/decompositions.py

```diff
@@ -1840,6 +1840,111 @@ def _native_batch_norm_legit_functional(
     return output, save_mean, save_rstd, new_running_mean, new_running_var
 
 
+def _get_batch_norm_reserve_tensor(
+    input: Tensor,
+    weight: Optional[Tensor],
+    bias: Optional[Tensor],
+    running_mean: Tensor,
+    running_var: Tensor,
+    eps: float,
+    cudnn_enabled: bool,
+) -> Tensor:
+    backend = torch._C._select_batch_norm_backend(  # type: ignore[attr-defined]
+        input, weight, bias, running_mean, running_var, True, eps, cudnn_enabled
+    )
+    if backend == torch._C._BatchNormBackend.Cudnn:  # type: ignore[attr-defined]
+        reserve_size = torch._C._get_cudnn_batch_norm_reserve_space_size(input)  # type: ignore[attr-defined]
+        return torch.empty(
+            reserve_size, dtype=input.dtype, layout=input.layout, device=input.device
+        )
+    else:
+        return Tensor()
```

Comment:
I'm not sure if our tracing infra will handle the plain `Tensor()` constructor very well - so I would just do `torch.empty(0)` here (one question is if this tensor should live on cpu or cuda - I guess if at this point we've already made the choice that we're not using cudnn in the first place, then leaving it on cpu is fine).

Reply:
Oh actually, I think we also saw that this tensor has `torch.byte` dtype